pkgs/development/tools/alire/default.nix

```diff
@@ -0,0 +1,36 @@
+{ stdenv, lib, fetchFromGitHub, gnat, gprbuild }:
+
+stdenv.mkDerivation {
+  pname = "alire";
+  version = "1.3-dev-2022-06-24";
+  src = fetchFromGitHub {
+    owner = "alire-project";
+    repo = "alire";
+    rev = "8f44a810";
+    hash = "sha256-kYntEMeVPo1+gRI+Q/6qfSyICXNf8yLT+LcbuyeRu/E=";
+    fetchSubmodules = true;
+  };
+
+  OS = if stdenv.isDarwin then "macOS" else null;
+
+  nativeBuildInputs = [ gnat gprbuild ];
+
+  buildPhase = ''
+    runHook preBuild
+    gprbuild -j0 -P alr_env
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    gprinstall --mode=usage -P alr_env -p -r --prefix=$out
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/alire-project/alire";
+    description = "Ada LIbrary REpository";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ ethindp ];
```

Comment:
Sorry for repeating, please add the platforms.
It seems to be limited to linux due to dependencies.

Reply:
I don't see the necessity of this. The package can be built on any platform; it is only the dependencies (gnatboot in particular) that limit where it can be built and run, and hopefully we can remove that barrier in the future. And I may already have a solution, though I can't test it. But I'll see about sending a PR sometime soon after this is approved.

Reply:
To clarify: `gnatboot` is the bootstrapping compiler, while `gnat`, which this package depends on, is the normal Ada compiler. It can target any platform GCC can target, so this is not actually a problem. We just need to configure cross-compiled builds of gnat on targets such as x86_64-darwin, and all will be fine. Linux can do the bulk of that, and if Linux is unable to build Mach-O executable images I can add in the native bootstrapping compiler for x86_64-darwin.